aio/aio-proxy/aio_proxy/search/es_search_runner.py

```diff
@@ -0,0 +1,150 @@
+            )
+        # If only filters are used, use nombre établissements ouverts to sort the
+        # results
+        else:
+            self.es_search_client = self.es_search_client.sort(
+                {"nombre_etablissements_ouverts": {"order": "desc"}},
+            )
+
+    def execute_and_format_es_search(self):
+        offset = self.search_params.page
+        page_size = self.search_params.per_page
+        es_search_client_with_aggr = self.es_search_client
+
+        self.es_search_client = self.es_search_client[offset : (offset + page_size)]
+        es_response = self.es_search_client.execute()
+        self.total_results = es_response.hits.total.value
+        self.execution_time = es_response.took
+
+        # Due to performance issues when aggregating on filter queries, we use
+        # aggregation on total_results only when total_results is lower than
+        # 10 000 results. If total_results is higher than 10 000 results,
+        # the aggregation causes timeouts on API. We return by default 10 000 results.
+        if self.total_results < MAX_TOTAL_RESULTS:
+            es_search_client_with_aggr.aggs.metric(
+                "by_cluster", "cardinality", field="siren"
+            )
+            es_search_client_with_aggr = es_search_client_with_aggr[
+                offset : (offset + page_size)
+            ]
+            es_search_client_with_aggr = es_search_client_with_aggr.execute()
+            self.total_results = (
+                es_search_client_with_aggr.aggregations.by_cluster.value
+            )
+            self.execution_time = es_search_client_with_aggr.took
+
+        self.es_search_results = []
+        for matching_unite_legale in es_response.hits:
+            matching_unite_legale_dict = matching_unite_legale.to_dict(
+                skip_empty=False, include_meta=False
+            )
+            # Add meta field to response to retrieve score
+            matching_unite_legale_dict["meta"] = matching_unite_legale.meta.to_dict()
+            # Add inner hits field (etablissements)
+            try:
+                matching_etablissements = (
+                    matching_unite_legale.meta.inner_hits.etablissements.hits
+                )
+                matching_unite_legale_dict["matching_etablissements"] = []
+                for matching_etablissement in matching_etablissements:
+                    matching_unite_legale_dict["matching_etablissements"].append(
+                        matching_etablissement.to_dict()
+                    )
+            except Exception:
+                matching_unite_legale_dict["matching_etablissements"] = []
+
+            self.es_search_results.append(matching_unite_legale_dict)
+
+    def sort_and_execute_es_search_query(self):
+        self.es_search_client = self.es_search_client.extra(track_scores=True)
+        self.es_search_client = self.es_search_client.extra(explain=True)
```

Comment:
Interesting, you always use explain true ? Does it have any impact on perf ?

Reply:
I always use it, but mainly for dev purposes. I will keep this in mind when refactoring queries.